DataLayer/MotorDetailsData/MotorDetailsData.cpp

```diff
@@ -1,12 +1,17 @@
 #include "MotorDetailsData.h"
 #include "MotorDetailsUnit.h"
 
-MotorDetailsData::MotorDetailsData(const unsigned char& numberOfMotors)
-    : motorDetailsUnits_(new MotorDetailsUnit[numberOfMotors])
+MotorDetailsData::MotorDetailsData(QList<I_MotorDetailsUnit*> units)
+    : motorDetailsUnits_(units)
 {
 }
 
-I_MotorDetailsUnit& MotorDetailsData::getMotorDetailsUnit(const unsigned char& index)
+unsigned char MotorDetailsData::getNumberOfUnits() const
```

Comment:
Why char instead of int? Will this cause an issue in for loops:
`for(int i = 0; i  < data.getNumberOfUnits(); i++` would there be an issue in comparing two different types?

Reply:
No it should be okay, I used unsigned char because the max value should be, so tried to conserve space?